### ISSUE TITLE
Hidden fields with dependencies disappear after setValue()

### ISSUE THREAD

**Issue by blauwaldt-it** [json-editor/json-editor#1521]
### General information

- json-editor version: 2.14.1

### Expected behavior

alt1 and alt2 are set correctly depending on switch also after loading a state using setValue()

### Actual behavior

after loading a state using setValue(json) only the altX is handled correctly, which was set in the json, the other (altY) has disappeared

### Steps to reproduce the behavior

1. load editor with schema below
2. depending on switch, editor.getValue('root') returns {"switch": true,"alt1": "1","alt2":undefined} OR {"switch": false,"alt1": undefined,"alt2":"2"}. This is expected and correct
3. Save state and load state using editor.setValue( editor.getValue('root') )
4. depending on switch, editor.getValue('root') now returns only {"switch": true,"alt1": "1"} or {"switch": false,"alt1": undefined}. "alt2" has disappeared (or only "alt2" is handled and "alt1" disppeared, depending on state when getting/setting state)

```javascript
{
    "title": "example",
    "type": "object",
    "properties": {

        "switch": {
            "title": "Switch",
            "type": "boolean",
            "format": "checkbox",
            "default": true
        },

        "alt1": {
            "title": "Alternative 1",
            "type": "string",
            "default": "1",
            "options": {
                "hidden": true,
                "dependencies": {
                    "switch": true
                }
            }
        },

        "alt2": {
            "title": "Alternative 2",
            "type": "string",
            "default": "2",
            "options": {
                "hidden": true,
                "dependencies": {
                    "switch": false
                }
            }
        }
    }
}```

**Comment by blauwaldt-it** [comment]
Seems to be the same as in #1452 and fixed with #1498

**Comment by blauwaldt-it** [comment]
Seems to be the same as in #1452 and fixed with #1498
